### DISCUSSION TITLE
Is there any (explicitly) way to dispose the SessionState after _core.Sessions.CreateSession?

### DISCUSSION THREAD

**Discussion by ingted** [NTDLS/Katzebase#91] [Q&A]
Only notice that TransactionReference implements IDisposable... @@"a

**Comment by NTDLS** [comment]
Yea, for general server functionality, sessions are long lived and are closed by the engine when the TCP/IP connection is closed. This is accomplished by a call to ```CloseByProcessId```.

```
var session = _core.Sessions.CreateSession(context.ConnectionId, param.Username, param.ClientName);

_core.Sessions.CloseByProcessId(session.ProcessId);
```

The close-on-disconnect is in: NTDLS.Katzebase.Server.APIService.cs

![image](https://github.com/user-attachments/assets/fd176396-d864-4916-8a4a-585c2d5d4f65)
